lib/web_ui/lib/src/engine/util.dart

```diff
@@ -54,115 +54,99 @@ String matrix4ToCssTransform(Matrix4 matrix) {
   return float64ListToCssTransform(matrix.storage);
 }
 
-/// Converts [matrix] to CSS transform value.
-String matrix4ToCssTransform3d(Matrix4 matrix) {
-  return float64ListToCssTransform3d(matrix.storage);
+/// Applies a transform to the [element].
+///
+/// See [float64ListToCssTransform] for details on how the CSS value is chosen.
+void setElementTransform(html.Element element, Float64List matrix4) {
+  element.style
+    ..transformOrigin = '0 0 0'
+    ..transform = float64ListToCssTransform(matrix4);
```

Comment:
Actually, this might be a good optimization, but we should probably evaluate several approaches, including setting it in a style sheet (the browser might be faster to apply it that us). There's also 13 other places where this is done, so we should probably find a unified solution. In this case specifically, the element is created outside the function, so each call site will have to make sure this property is set.

I filed https://github.com/flutter/flutter/issues/50358 for us to revisit this across the entire codebase.